include/utils.hpp

```diff
@@ -37,9 +37,113 @@ namespace utils {
+                            2.65,
+                            2.98,
+                            3.34,
+                            3.72,
+                            4.14,
+                            4.58,
+                            5.06,
+                            5.57,
+                            6.11,
+                            6.68,
+                            7.30,
+                            7.94,
+                            8.63,
+                            9.35,
+                            10.12,
+                            10.92,
+                            11.77,
+                            12.66,
+                            13.60,
+                            14.59,
+                            15.62,
+                            16.70,
+                            17.83,
+                            19.02,
+                            20.26,
+                            21.55,
+                            22.90,
+                            24.30,
+                            25.77,
+                            27.23,
+                            28.70,
+                            30.16,
+                            31.63,
+                            33.09,
+                            34.56,
+                            36.02,
+                            37.49,
+                            38.95,
+                            40.41,
+                            41.88,
+                            43.34,
+                            44.81,
+                            46.27,
+                            47.74,
+                            49.20,
+                            50.67,
+                            52.13,
+                            53.60,
+                            55.06,
+                            56.53,
+                            57.99,
+                            59.45,
+                            60.92,
+                            62.38,
+                            63.85,
+                            65.31,
+                            66.78,
+                            68.24,
+                            69.71,
+                            71.17,
+                            72.64,
+                            74.10,
+                            75.57,
+                            77.03,
+                            78.50,
+                            79.96,
+                            81.42,
+                            82.89,
+                            84.35,
+                            85.82,
+                            87.28,
+                            88.75,
+                            90.21,
+                            91.68,
+                            93.14,
+                            94.61,
+                            96.07,
+                            97.54,
+                            99.00
+    };
+
+    float calc = (current * 100) / (float)total;
+
+    for (int r=0; r<100; r++) {
+      if (calc <= rank_coefs[r]) return (uint64_t)r;
```

Comment:
minor detail but since we return int anyway, would be better to just have a static table of [int] in the first place.

Reply:
How do you propose to do the roundings? Just round down everything?

Reply:
No just math-round... >0.5 is +1